lab_1/main.py

```diff
@@ -86,10 +146,41 @@ def compare_profiles_advanced(unknown_profile: dict, profile_to_compare: dict, t
     :return: a dictionary with 7 keys – name, score, common, sorted_common, max_length_word,
     min_length_word, average_token_length
     """
-    pass
-
-
-def detect_language_advanced(unknown_profile: dict, profiles: list, languages: list, top_n: int) -> str or None:
+    if isinstance(unknown_profile, dict) and isinstance(profile_to_compare, dict)\
+        and isinstance(top_n, int):
+        lang_adv = {}
+        cross = []
+        lang_adv['name'] = profile_to_compare['name']
+        for i in get_top_n_words(profile_to_compare['freq'], top_n):
+            if i in get_top_n_words(unknown_profile['freq'], top_n):
+                cross.append(i)
+            else:
+                continue
+        length = 0
+        for words in profile_to_compare['freq']:
+            length += len(words)
+        max_word = 0
+        min_word = 0
+        for max_length in profile_to_compare['freq']:
```

Comment:
do you iterate over lengths or words or freqs? naming is a bit confusing

Reply:
I iterate over words in the freq dictionary from profile_to_compare, as that is where I keep words in the language profile

Reply:
Yep, but I can't understand that from the variable name
better to name like `word`